Initialize tensor with fp32 in Caffe2Backend.prepare()

yinghai opened the pull request:
Summary: Fix https://github.com/pytorch/pytorch/issues/14104

Differential Revision: D13598332

bddppq commented on caffe2/python/onnx/backend.py:
```diff
@@ -644,7 +644,7 @@ def _direct_initialize_inputs(cls, inputs, initialized, ws, device_option):
                 continue
             shape = list(d.dim_value for d in value_info.type.tensor_type.shape.dim)
-            ws.FeedBlob(value_info.name, np.ones(shape), device_option)
+            ws.FeedBlob(value_info.name, np.ones(shape, dtype=np.float32), device_option)
```
value_info should have the actual dtype

houseroad reviewed (approved):
LGTM, nit: add some test to cover this code

bddppq commented on caffe2/python/onnx/backend.py:
```diff
@@ -644,7 +644,10 @@ def _direct_initialize_inputs(cls, inputs, initialized, ws, device_option):
-            ws.FeedBlob(value_info.name, np.ones(shape), device_option)
+            ws.FeedBlob(
+                value_info.name,
+                np.ones(shape, dtype=onnx.mapping.TENSOR_TYPE_TO_NP_TYPE[value_info.type]),
```
`value_info.type.elem_type`

yinghai commented:
@pytorchbot restest this please

yinghai commented:
@pytorchbot retest this please

Commit:
Initialize tensor with fp32 in Caffe2Backend.prepare() (#15832)

Summary:
Fix https://github.com/pytorch/pytorch/issues/14104
Pull Request resolved: https://github.com/pytorch/pytorch/pull/15832

Reviewed By: bddppq

Differential Revision: D13598332

Pulled By: bddppq

fbshipit-source-id: ab38169095b29b9f29ed52d0b58a764d2f56539f
